app/navigation/nonsipp/SharesNavigator.scala

```diff
@@ -213,120 +213,203 @@ object SharesNavigator extends JourneyNavigator {
   val checkRoutes: UserAnswers => UserAnswers => PartialFunction[Page, Call] = _ =>
     userAnswers => {
 
+      case TypeOfSharesHeldPage(srn, index) =>
+        userAnswers.get(TypeOfSharesHeldPage(srn, index)) match {
+          case Some(SponsoringEmployer) =>
+            controllers.nonsipp.shares.routes.WhyDoesSchemeHoldSharesController.onPageLoad(srn, index, CheckMode)
+          case Some(Unquoted) =>
+            controllers.nonsipp.shares.routes.WhyDoesSchemeHoldSharesController.onPageLoad(srn, index, CheckMode)
+          case Some(ConnectedParty) =>
+            controllers.nonsipp.shares.routes.SharesCYAController.onPageLoad(srn, index, CheckMode)
```

Comment:
For all these navigation changes in check routes, do we need to add corresponding unit tests in SharesNavigatorSpec ?

Reply:
Yes, we need to fix @cyeniceri. Let me check if I can fix in this or next one